src/internal/modules/billing/pre-handlers.js

```diff
@@ -25,5 +27,28 @@ const redirectToWaitingIfEventNotComplete = async (request, h) => {
     : h.redirect(`/waiting/${event.event_id}`).takeover();
 };
 
+const checkBatchStatus = async (request, h, status) => {
+  const { batch } = request.pre;
+  if (batch.status !== status) {
+    return Boom.badRequest(`Batch ${batch.id} has unexpected status ${batch.status}`);
```

Comment:
I wonder if this is actually a bad request. It feels like the request was valid and understood, but the server is refusing to do anything with it. This might be a 403 forbidden where the reason stays present and is therefore not confused with an authentication/authorisation error.

Difficult to know, it is a shame there is not a code for 

1) I don't understand your request
2) I understand the request but it is against the rules
3) I understand the request but you are not allowed to action it